internal-tool frameworks blog (#1650)
pankod/refine: documentation/blog/2022-03-16-internal-tool-frameworks.md

slug: best-internal-tool-react-frameworks
authors: melih
tags:
    [
        refine,
        react,
        admin-panel,
        business-tool,
        internal-tool,
        B2B,
        B2C,
        open-source,
    ]
image: https://refine.dev/img/refine_social.png
hide_table_of_contents: false
---

import superplate from '@site/static/img/blog/2022-03-16-internal-tool-frameworks/superplate.gif';
import refine_admin_panel from '@site/static/img/blog/2022-03-16-internal-tool-frameworks/refine-admin-panel.png';
import refine_client from '@site/static/img/blog/2022-03-16-internal-tool-frameworks/refine_client.png';
import redwood from '@site/static/img/blog/2022-03-16-internal-tool-frameworks/redwood.png';
import react_admin from '@site/static/img/blog/2022-03-16-internal-tool-frameworks/react_admin.png';

## Best Internal Tool Frameworks for ReactJS

In this article, I will talk about new open-source React.js frameworks that will greatly contribute to your project while developing internal-tool applications and make your work easier with their features.

<!--truncate-->

## What is Internal Tool?

> Internal tools, also known as Back Office applications, are software pieces developed to improve your corporate processes and make your work easier. Internal tool applications are generally used to control your business follow-up or to manage your product's internal processes.

Reviewer: ```suggestion
Internal tools, also known as Backoffice applications, are software pieces developed to improve your corporate processes and make your work easier. Internal tool applications are generally used to control your business follow-up or to manage your product's internal processes.
```

Internal tools manage the activities your company or product needs, such as tracking inventory, processing customer requests, controlling payments. Although its field is quite wide, the needs of each product or company differ. Here in this article, we will review React.js Frameworks that will help you quickly develop your own internal-tool applications according to your needs.

> Interal Tool React.js Frameworks that we will review:

Reviewer: ```suggestion
Internal Tool React.js Frameworks that we will review:
```

📦 Out-of-the-box : Routing, networking, authentication, state management, i18n and UI.

🔌 Backend Agnostic : Connects to any custom backend. Built-in support for REST API, GraphQL, NestJs CRUD, Airtable, Strapi, Strapi v4, Strapi GraphQL, Supabase, Hasura, Nhost, Appwrite, Firebase, Directus and Altogic.

📝 Native Typescript Core : You can always opt out for plain JavaScript.

🐜 Enterprise UI : Works seamlessly with Ant Design System. (Support for multiple UI frameworks is on the Roadmap)

📝 Boilerplate-free Code : Keeps your codebase clean and readable.

### Installation

You can quickly create the project containing **refine** and all the features you will use with it, thanks to **refine**'s Project Creator CLI (Superplate). No extra setup required!

<div class="img-container">
    <div class="window">
        <div class="control red"></div>
        <div class="control orange"></div>
        <div class="control green"></div>
    </div>
    <img src={superplate} alt="Refine Project Creator CLI" />
</div>
<br />

If you want, you can manually install the refine packages.

```bash

> npm i @pankod/refine-core
> 
> npm i @pankod/refine-antd

Reviewer: ```suggestion
npm i @pankod/refine-core @pankod/refine-antd
```

---

> title: ReactJS Frameworks You Should Know Before You Start Developing Internal Tool Application

Reviewer: ```suggestion
title: ReactJS Frameworks You Should Know Before Start Developing B2B/Internal Application
```
